client/my-sites/patterns/hooks/use-pattern-search-term.ts

```diff
@@ -24,7 +24,7 @@ export function usePatternSearchTerm(
 		}
 
 		if ( url.href !== location.href ) {
-			page.redirect( url.href );
+			page( url.href.replace( url.origin, '' ) );
```

Comment:
This isn't directly related to the core changes here, it's just another follow-up to #88614 that fixes an issue where the page is reloaded every time you do a search 😬